### PR TITLE
Configuration as Code compatibility tests

### PR DESCRIPTION
**Pull request by MRamonLeon**
I've added to the plugin the tests already done on [Configuration as Code plugin](https://github.com/jenkinsci/configuration-as-code-plugin) to test the JCasC compatibility on each release.

I've added some documentation on how to configure the plugin with CasC.

@rsandell @varyvol @alecharp @batmat

### CODE REVIEW

**Comment by MRamonLeon** [comment]
@rsandell do you agree to merge this PR if it's fine to you? Thanks!

**Comment by MRamonLeon** [comment]
The error on CI seems infra related, I added it to this ticket, which seems to be related: https://issues.jenkins-ci.org/browse/JENKINS-55287

File: pom.xml [hunk 1]
```diff
@@ -234,7 +242,21 @@
     <dependency>
       <groupId>org.jsoup</groupId>
       <artifactId>jsoup</artifactId>
-      <version>1.10.2</version>
+      <version>1.11.3</version>
+      <scope>test</scope>
```
Comment (rsandell) [inline pom.xml]: So this wasn't in the test scope before, what prompted you to put it there now?

Comment (MRamonLeon) [reply pom.xml]: I've only changed the version, it was already on the test scope. I did so because JCasC uses this version.

Comment (rsandell) [reply pom.xml]: Ah, sorry, the diff was a bit weird :)

Comment (MRamonLeon) [reply pom.xml]: yes, it was

**Pull request merged**